### PR TITLE
Add mycusini 2.0

### PR DESCRIPTION
**Pull request by sasosmolej**
Adds:
- Print4Taste vendor (brands/product families mycusini and procusini)
- Filament settings for choco cartridges (proprietary)
- 3 basic profiles, similar to the included models (solid, outlines and fillables)

Known issues:
- Reported bed size may not be fully usable - requires further testing
- Time estimates are inaccurate - for example 21 minute print is off by 1,5 minutes - though at limited cartridge sizes that's about as off as it gets
- Usage is reported in grams while remaining in printer is reported as % remaining - how much a reported percent is in extruder mm needs further testing
- Current auto cooling (fanless, just time based) may be insufficient for actual 3D sculptures and could be further improved by experimentation
- Layer change retracts/wipes need tuning to avoid gaps in perimeters
- "Filament runout" triggers a full stop of the print - unknown if this can be avoided with gcode
- Thumbnails are not generated since Prusaslicer does not support M10086/gimage/simage thumbnails

Additional notes:
- Consider these settings an ALPHA version
- These settings are faster than printer defaults, but with sufficient full preheat, extrusion works perfectly fine
- Bed model is just a rectangle - it's only so that the bed texture is displayed
- Volumetric slope is used buy may not be required
- Using a lower extrusion than nozzle diameter results in material clumping, which exacerbates gaps in perimeters/infill - all extrusion widths may require further tuning or reducing layer heights
- Making hollow parts with top surfaces is just very, very difficult, or it uses a lot more material than it should

image

### CODE REVIEW

File: live/Print4Taste/0.0.1.ini [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+top_infill_extrusion_width = 1.15
+top_solid_infill_speed = 100%
+top_solid_layers = 3
+travel_speed = 80
+skirts = 0
+
+
+[print:0.50mm SOLID @MC2.0]
+inherits = *common_MC2.0*
+alias = 0.50mm SOLID
```
Comment (rtyr) [inline live/Print4Taste/0.0.1.ini:78]: alias is not needed (we are hiding part behind @ automatically)

File: live/Print4Taste/0.0.1.ini [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+alias = 0.50mm OUTLINES
+bottom_solid_layers = 0
+
+
+[filament:mycusini 3D Choco @MC2.0]
+alias = mycusini 3D Choco
+bed_temperature = 0
+bridge_fan_speed = 0
+compatible_printers_condition = printer_notes=~/.*PRINT4TASTE.*/ and printer_notes=~/.*MYCUSINI2.0.*/
+compatible_prints_condition = printer_notes=~/.*PRINT4TASTE.*/ and printer_notes=~/.*MYCUSINI2.0.*/
```
Comment (rtyr) [inline live/Print4Taste/0.0.1.ini:99]: Invalid condition. I removed this line as it is not needed.

File: live/Print4Taste/0.0.1.ini [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+fill_density = 0%
+spiral_vase = 1
+
+[print:0.50mm OUTLINES @MC2.0]
+inherits = 0.50mm FILLABLE @MC2.0
+alias = 0.50mm OUTLINES
+bottom_solid_layers = 0
+
+
+[filament:mycusini 3D Choco @MC2.0]
```
Comment (rtyr) [inline live/Print4Taste/0.0.1.ini:94]: added filament_vendor

File: live/Print4Taste/index.idx [hunk 4]
```diff
@@ -0,0 +1,2 @@
+min_slic3r_version = 2.5.0
```
Comment (rtyr) [inline live/Print4Taste/index.idx:1]: changed to 2.6.0-alpha1

**Review by rtyr — COMMENTED** [review]
I also renamed resources to match ini file (mc2.0 to MC2.0), some operating systems are more case sensitive.

**Comment by sasosmolej** [comment]
Thanks for the corrections and quick merge.

Note that cartridges are specifically compatible only with this printer model (maybe for mycusini 1.0?), but not with others (procusini has a different diameter for example).
Not sure if that requires changing anything, but it's the reason why I put the printer model compatibility to the filament.

**Comment by rtyr** [comment]
> Not sure if that requires changing anything, but it's the reason why I put the printer model compatibility to the filament.

That is OK, compatible printer condition is still there. I only removed compatible print condition (which was incorrect anyway). You can run `prusa-slicer-console.exe --loglevel=5` to see the issues in the bundle.

**Comment by rtyr** [comment]
There is only one printer at the moment, we can review/modify it again when new printer model will be added.

**Comment by rtyr** [comment]
At the moment we can add new printers only with PrusaSlicer release (because of thumbnail and other resources), but material profiles can be added at any time via on-line update.

**Pull request closed without merging**